### ISSUE TITLE
Certain Samples Cause Extreme Training Step Latency

### ISSUE THREAD

**Issue by YiyuHong** [bytedance/Protenix#302]
**Environment**
- GPUs: 8× NVIDIA B200
- Training mode: Single node, Multi-GPU
- Training config: Default Protenix settings and hyperparameters using `train_demo.sh` (layernorm, triangle all set torch)
- pytorch 2.11, cuda 13.0

**Problem**

During training, some samples take an extremely long time to process — sometimes taking an hour or more per sample. According to the Protenix report and documentation, the expected throughput is approximately **12 seconds/step**, but in practice certain samples takes **1 hour/step**.

Because training is synchronous across all GPUs, the slow sample on one GPU forces all other GPUs to idle and wait, causing an effective training stall.

Current workaround: increasing the NCCL timeout prevents the job from crashing, but does not resolve the underlying latency issue, and some samples still stall even with a very large timeout.

```python
timeout_seconds = int(os.environ.get("NCCL_TIMEOUT_SECOND", 4800))
dist.init_process_group(
    backend="nccl", timeout=datetime.timedelta(seconds=timeout_seconds)
)
```

**Question**

- Has anyone encountered the same issue?
- Is there a known class of samples (e.g., large MSAs, complex assemblies, difficult spatial crops) that triggers this behavior?
- Is there a recommended way to filter or cap these samples before they enter the training pipeline?

Any guidance on root cause or workaround would be appreciated.

**Comment by YiyuHong** [comment]
It seems the crop stage is spending too much time, especially on the template generating part.
```
template_features = DataPipeline.get_template_raw_features(
    bioassembly_dict=bioassembly_dict,
    selected_indices=selected_indices,
    template_featurizer=template_featurizer,
)
```

**Comment by YiyuHong** [comment]
just install kalign and hmmer solve the problem.
